pkg/odo/cli/service/operator_backend.go

```diff
@@ -240,3 +253,42 @@ func (b *OperatorBackend) DeleteService(o *DeleteOptions, name string, applicati
 
 	return nil
 }
+
+func (b *OperatorBackend) buildCRDfromParams(o *CreateOptions, csv olm.ClusterServiceVersion) (map[string]interface{}, error) {
+	var cr *olm.CRDDescription
+	hasCR := false
+	CRs := o.KClient.GetCustomResourcesFromCSV(&csv)
+	for _, custRes := range *CRs {
+		c := custRes
+		if c.Kind == b.CustomResource {
+			cr = &c
+			hasCR = true
+			break
+		}
+	}
+	if !hasCR {
+		return nil, fmt.Errorf("the %q resource doesn't exist in specified %q operator", b.CustomResource, o.ServiceType)
+	}
+
+	crBuilder := service.NewCRBuilder(cr)
+	var errorStrs []string
+
+	for key, value := range o.ParametersMap {
+		err := crBuilder.SetAndValidate(key, value)
+		if err != nil {
+			errorStrs = append(errorStrs, err.Error())
+		}
+	}
```

Comment:
Shouldn't the code in business logic build the CRD with the help of parameters provided by the user instead of doing it in cli package? So, here, instead of iterating over `o.ParametersMap`, I think we shuold pass the parameters to business logic and let it just return us a CRD or error.

Building a CRD from params (going by the name of this function - `buildCRDfromParams`) should be done at business logic.

Reply:
when you are saying "at business logic" where would you suggest I create the `buildCRDfromParams`?

Reply:
ok, I mostly figured out what was needed to done and applied the change, please take a look @dharmit .